ext/adaptors/ranges-lib.tex

```diff
@@ -659,7 +678,8 @@
 
 \indexlibrary{\idxcode{next}!\idxcode{subrange}}%
 \begin{itemdecl}
-[[nodiscard]] constexpr subrange next(difference_type_t<I> n = 1) const;
+[[nodiscard]] constexpr subrange next(difference_type_t<I> n = 1) const
+  requires ForwardIterator<I>;
```

Comment:
Why?

Reply:
To avoid spooky action at a distance. This copies the iterator and increments the copy; if `I` is single-pass, calling `next` will invalidate `*this`. Other options are (1) have `next` be non-`const` when `I` is single-pass, or (2) I just get over the spooky action at a distance and we add a note telling users that calling `next` invalidates single-pass subranges.

Reply:
My vote goes for (2). It's true generally that operations on input ranges are not equality-preserving.

Reply:
There's a bit of a safety argument that users should call `advance` instead of `next` with input ranges to avoid having invalidated ranges lying around. I'll keep it in mind as we gather usage experience for `subrange`, but revert this for now.